values.yaml

```diff
@@ -49,7 +49,8 @@ st2:
   #password: Ch@ngeMe
   # ST2 crypto key for the  K/V datastore.
   # See https://docs.stackstorm.com/datastore.html#securing-secrets-admin-only for more info.
-  # Warning! Replace with your own generated key!
+  # If set, st2.datastore_crypto_key always overrides any existing datastore_crypto_key.
+  # If not set, the datastore_crypto_key is auto-generated on install and preserved across upgrades.
   #datastore_crypto_key: >-
   #  {"hmacKey": {"hmacKeyString": "", "size": 256}, "size": 256, "aesKeyString": "", "mode": "CBC"}
   # SSH private key for the 'stanley' system user ('system_user.ssh_key_file' in st2.conf)
```

Comment:
How to configure the chart to disable the use of the datastore crypto key?

Reply:
Why would you want to disable the datastore crypto key?

Reply:
That's how the previous configuration was before this PR.

Reply:
Would setting `datastore_crypto_key: disable` be ok as a way to disable this?

Reply:
I'm looking at the `st2.conf` example  https://github.com/StackStorm/st2/blob/master/conf/st2.conf.sample#L176-L177
If the `encryption_key_path =` is empty, stackstorm just won't use it.

I think if we'll follow that way it would be simple enough without an additional setting.
Would also make sense to document the empty option in the Helm values comments.

Reply:
If `encryption_key_path` is not defined in st2.conf, won't StackStorm refuse to encrypt any datastore entries? ie nothing would have to be re-encrypted since nothing CAN be encrypted without an encryption key. So, I don't think adding an encryption key to existing installations (where none is currently provided) would be a problem. Do I have a faulty assumption somewhere in here?

As far as making it possible to disable the encryption key: This PR auto-generates the datastore_crypto_key if it is not defined. How can I distinguish between the new default (datastore_crypto_key value is empty so auto-generate it) and the old default (default_crypto_key value is empty, so set encryption_key_path to empty as well). Maybe I should flag this as a breaking change in the changelog? I think this is backwards compatible, as described in the last paragraph.

Reply:
distinguishing between `# datastore_crypto_key` (unset) and `datastore_crypto_key: ""` (empty string) makes the template more complex and feels hacky: https://github.com/Masterminds/sprig/issues/53#issuecomment-483414063

It's complex enough that I'd probably need to introduce a helper just to determine whether or not to create or mount the crypto_key secret.

Reply:
Thanks for researching this, the new flag would be nice then per your initial idea :+1: 

And yes, in any way this is a breaking change as people who had no st2 K/V crypto key configured in Helm values would find that it'll be forcefully autogenerated for them after the upgrade, including the change in st2.conf.

Reply:
done